File: programmers/Lv_2/KimHyunSeo/기능개발.cpp
#include <iostream>
#include <string>
#include <vector>
#include <cmath>

using namespace std;
vector<int> solution(vector<int> progresses, vector<int> speeds)
{
	vector<int> temp;
	vector<int> answer;
	int max = 0;
	int count = 1;
	//100=progresse[0]+speeds[0]*x
	for (int i = 0; i < progresses.size(); i++){
		double x = (100 - (double)progresses[i]) / (double)speeds[i];
		int y=ceil(x);
		temp.push_back(y);  
	}
	max = temp[0];//7
	for (int i = 1; i < temp.size(); i++)
	{	
		if (temp[i] <= max)
			count++;
		else {
			answer.push_back(count);
			count = 1;
			max = temp[i];
		}
	}
	answer.push_back(count);
	return answer;
}

int main()
{
	vector<int> progresses = { 93,30,55,95,90 };  //7,3,9,4,10
	vector<int> speeds = { 1,30,5,4,1 };
	solution(progresses, speeds);
}

File: programmers/Lv_2/Kimyongho/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue <Integer> q = new LinkedList<Integer>();
		for(int i = 0; i < progresses.length; i++) {
			int time = (100 - progresses[i])/speeds[i];
			q.offer(time);
		}
		Queue <Integer>q2 = new LinkedList<Integer>();
		while(!q.isEmpty()) {
			int count = 1;
			int big = q.poll();
			while(!q.isEmpty() && big >= q.peek()) {
				count++;
				q.poll();
			}
			q2.offer(count);
		}
		int [] answer = new int[q2.size()];
		int index = 0;
		while(!q2.isEmpty()) {
			answer[index++] = q2.poll();
		}
        return answer;
    }
}
